### ISSUE TITLE
[BUG] Item.give() doesn't work because of the new format

### ISSUE THREAD

**Issue by patrykduda2005** [WingedSeal/jmc#111]
## Describe the bug
Item.give(...); puts its nbts in old format

## To Reproduce
1. write proper Item.create() and Item.give()
2. compile

## Expected behavior
nbt should be put in "[]" instead of "{}" and values should use "=" instead of  ":"

## Screenshots
![image](https://github.com/user-attachments/assets/0887c16b-c813-4d63-84ba-543145d2d312)
![image](https://github.com/user-attachments/assets/12ebe81e-90b4-4473-ba52-b1b7894402b7)
![image](https://github.com/user-attachments/assets/5393acb9-c8c9-4fc8-8ccb-4bc2ba5de938)

## Desktop
- Linux

**Comment by Nico314159** [comment]
Yeah, this feature is very hard to update. I'll try to see what I can do, but I can't make any promises because I have a lot of other things I need to do. For now, I have two workarounds.

## Option 1: `#define`

In an `.hjmc` file, type 
```h 
#define dummy_cooked_cod enchanted_book[max_stack_size=64,custom_name='"Cooked Cod"',lore=['"DUMMY"'],rarity="common",custom_model_data=421]
```
Then, if you type
```mcfunction
give @a dummy_cooked_cod 10;
```
in your main file, it will give you the expected result.



## Option 2: Loot Tables

Use the following syntax to create a loot table:
```ts
new loot_table(dummy_cooked_cod) {
	"pools": [
		{
			"rolls": 1,
			"entries": [
				{
					"type": "minecraft:item",
					"name": "minecraft:enchanted_book",
					"functions": [
						{
							"function": "minecraft:set_name",
							"name": "Cooked Cod"
						},
						{
							"function": "minecraft:set_lore",
							"lore": ["DUMMY"],
							"mode": "replace_all"
						},
						{
							"function": "minecraft:set_count",
							"count": 10
						},
						{
							"function": "minecraft:set_components",
							"components": {
								"minecraft:rarity": "common",
								"minecraft:max_stack_size": 64
							}
						}
					]
				}
			]
		}
	]
}
```
Then, you can use `loot give @a loot namespace:dummy_cooked_cod;` anywhere in your code and it will have the result you wanted above. I actually recommend loot tables a lot, because they can do powerful things that the `item` command simply cannot (ex: if you wanted to make the count based on some score or variable instead of a fixed number, you can do that.)

**Comment by Dufftwt** [comment]
The workarounds are great, but I think it's time to implement this, it's been since last year this is open and still is a major issue

**Comment by Nico314159** [comment]
> The workarounds are great, but I think it's time to implement this, it's been since last year this is open and still is a major issue

I understand that you care about this issue, but implementing it is easier said than done. If it were a simple task, it would have already been done.

Besides, I generally would advise against using the `give` command in the first place when the `loot` command is simply better.

**Comment by WingedSeal** [comment]
fixed in https://github.com/WingedSeal/jmc/commit/7087ce1471a6f011a62198c5a1f52f2fbe181280

@Nico314159 
> Besides, I generally would advise against using the `give` command for complex items in the first place when the `loot` command is simply better.

remind me to make Item.createTable or something

> I understand that you care about this issue, but implementing it is easier said than done. If it were a simple task, it would have already been done.

thank you for sugar coating my laziness lol
https://www.youtube.com/watch?v=-pv8D-ydR1s


@Dufftwt 
> The workarounds are great, but I think it's time to implement this, it's been since last year this is open and still is a major issue

here you go
fixed now
at least i think so
still expecting tons of bugs
